src/main/java/seedu/address/model/person/Room.java

```diff
@@ -21,15 +25,33 @@
      */
     public Room(String room) {
         requireNonNull(room);
-        checkArgument(isValidRoom(room), MESSAGE_CONSTRAINTS);
+        checkArgument(isValidRoom(room), getMessageConstraints());
         value = room;
     }
 
     /**
      * Returns true if a given string is a valid Room number.
      */
     public static boolean isValidRoom(String test) {
-        return test.matches(VALIDATION_REGEX);
+        if (!test.matches(VALIDATION_REGEX)) {
+            return false;
+        }
+        int room = Integer.parseInt(test);
+        int roomNo = room % 100;
+        int floorNo = Math.floorDiv(room, 100);
+        boolean isRoomValid = (roomNo <= MAX_ROOM_NUMBER) && (roomNo >= MIN_ROOM_NUMBER);
+        boolean isFloorValid = (floorNo <= MAX_FLOOR_NUMBER) && (floorNo >= MIN_FLOOR_NUMBER);
+        return isFloorValid && isRoomValid;
+    }
+
+    public static String getMessageConstraints() {
+        return MESSAGE_CONSTRAINTS + "\nFloor : " + MIN_FLOOR_NUMBER + " to " + MAX_FLOOR_NUMBER
+                + "\nRoom: " + MIN_ROOM_NUMBER + " to " + MAX_ROOM_NUMBER;
+    }
+
+    public static void setRoomPref(int pref) {
+        MAX_FLOOR_NUMBER = Math.floorDiv(pref, 100);
+        MAX_ROOM_NUMBER = pref % 100;
```

Comment:
Can consider adding set `MIN_FLOOR_NUMBER` and `MIN_ROOM_NUMBER`, e.g. level 1 is the common lounge.